packages/daemon/src/rpc/methods/app.js

```diff
@@ -111,3 +133,32 @@ export const setPermission = async (
     await ctx.openVault.save()
   }
 }
+
+const constructClientSession = (
+  ctx: RequestContext,
+  appID: ID,
+  userID: ID,
+  session: SessionData,
+): ClientSession => {
+  const user = ctx.openVault.identities.getOwnUser(userID)
+  const app = ctx.openVault.apps.getByID(appID)
+  if (user && app) {
+    return {
+      user: {
+        id: userID,
+        data: user.data,
+      },
+      session: {
+        id: session.sessID,
+        permissions: session.permissions,
+      },
+      app: {
+        id: appID,
+        manifest: app.manifest,
+      },
+    }
+  } else {
+    const type = !user ? 'user' : 'app'
+    throw new Error(`Missing ${type}`)
```

Comment:
Rather than having `else/if` statements and additional logic with the error, you could throw earlier in the function body, ex:

```js
const app = ...
if (app == null) {
  throw new Error('Invalid appID')
}
const user = ...
if (user == null) {
  throw new Error('Invalid userID')
}
return ...
```